### DISCUSSION TITLE
SubscriptionWorker run cancellation when "no docs left" doesn't seem to work

### DISCUSSION THREAD

**Discussion by anemitoff** [ravendb/ravendb#12373] [Q&A]
I have a subscription worker with a subscription that at some point temporarily has no docs left to process (due to filter on `'@metadata'.'@refresh'` that I set on batch item handling to retry after a time interval).

I have a test that I arbitrarily want to end after 4 seconds.  

If the cancellation token used for `await worker.Run(batch=>{}, cancellationToken)` is set after 4 seconds and the subscription is "idle" because all documents are filtered out, the await is never satisfied. 

Any idea on how to fix?  
If not, I will submit test later.

**Comment by garayx** [comment]
Hi 

Please send the test
